src/core/server/saved_objects/service/lib/collect_multi_namespace_references.ts

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { ISavedObjectTypeRegistry } from '../../saved_objects_type_registry';
+import type { SavedObjectsSerializer } from '../../serialization';
+import type { SavedObject, SavedObjectsBaseOptions } from '../../types';
+import { getRootFields } from './included_fields';
+import { getSavedObjectFromSource, rawDocExistsInNamespace } from './internal_utils';
+import type { RepositoryEsClient } from './repository_es_client';
+
+/**
+ *
+ * @public
+ */
+export interface SavedObjectsCollectMultiNamespaceReferencesObject {
+  id: string;
+  type: string;
+}
+
+/**
+ *
+ * @public
+ */
+export interface SavedObjectsCollectMultiNamespaceReferencesOptions
+  extends SavedObjectsBaseOptions {
+  /** Whether or not to include tags when collecting references */
+  excludeTags?: boolean;
```

Comment:
> I think I'd rather have the security plugin hard-code this than to introduce an entirely new wrapper, but I'm willing to be convinced otherwise. I'd be interested to see what the core team prefers here.

Well, we want this to behave the same way whether Security is enabled or not. So we would have to hard-code this at the SOR level. I 100% agree that the additional wrapper is egregious but I'm not sure what the best alternative is. I'm also interested to see what the Core team prefers.

Reply:
Ah makes sense. Now that we no longer ship an OSS distribution, I wonder if there are pieces of the tagging plugin that can be integrated into the core saved objects service -- this could become a "core type", for example, much like `config` is today. Agree, let's see what Core team prefers.

Reply:
> I know this is poor isolation between the SOM ui and the tagging feature, but if that avoid adding this new SO client wrapper only to toggle an option, it seems like totally worth it.

++